Source/Engine/AGS.Engine/Graphics/Animations/AGSAnimationComponent.cs

```diff
@@ -34,11 +32,16 @@ public void StartAnimation(IAnimation animation)
                 scale.BaseSize = new SizeF(animation.Frames[0].Sprite.Width, animation.Frames[0].Sprite.Height);
 			}
 			IAnimation currentAnimation = Animation;
-			currentAnimation?.State.OnAnimationCompleted.TrySetResult (new AnimationCompletedEventArgs (false));
-			
-			Animation = animation;
+            if (currentAnimation != null)
+            {
+                currentAnimation.State.OnNextFrame.Subscribe(OnAnimationFrame);
+                currentAnimation.State.OnAnimationCompleted.TrySetResult(new AnimationCompletedEventArgs(false));
```

Comment:
This is a mistake: currentAnimation holds the currently playing animation, the one we're about to replace with the new animation. So you should not subscribe to currentAnimation here, you should unsubscribe currentAnimation and subscribe to animation -> this probably explains the glitches you've seen.